hail/python/hail/table.py

```diff
@@ -3641,49 +3641,56 @@ def _same(self, other, tolerance=1e-6, absolute=False, reorder_fields=False):
             print(f'Table._same: types differ:\n  {self._type}\n  {other._type}')
             return False
 
-        left_global_value = Env.get_uid()
-        left_value = Env.get_uid()
         left = self
-        left = left.select_globals(**{left_global_value: left.globals})
-        left = left.group_by(_key=left.key).aggregate(**{left_value: hl.agg.collect(left.row_value)})
+        left = left.select_globals(left_globals = left.globals)
```

Comment:
These don't clash with anything because we're throwing all the globals under it. Same with the rows & row keys.